robot/packages/utils.go

```diff
@@ -257,45 +264,160 @@ func commonCleanup(logger logging.Logger, expectedPackageEntries map[string]bool
 	for _, packageTypeDir := range topLevelFiles {
 		packageTypeDirName, err := rutils.SafeJoinDir(packagesDataDir, packageTypeDir.Name())
 		if err != nil {
+			logger.Debugf("Failed to join directory name %s: %v", packageTypeDir.Name(), err)
 			allErrors = errors.Join(allErrors, err)
 			continue
 		}
 
-		// Delete any non-directory files in the packages/data dir except for those with suffixes:
-		//
-		// `.status.json` - these files contain download status infomration.
-		// `.first_run_succeeded` - these mark successful setup phase runs.
-		if packageTypeDir.Type()&os.ModeDir != os.ModeDir && !strings.HasSuffix(packageTypeDirName, statusFileExt) {
-			allErrors = errors.Join(allErrors, os.Remove(packageTypeDirName))
+		// Handle regular files (non-directories)
+		if packageTypeDir.Type()&os.ModeDir != os.ModeDir {
+			// Skip status files and first run success files
+			if strings.HasSuffix(packageTypeDirName, statusFileExt) ||
+				strings.HasSuffix(packageTypeDirName, config.FirstRunSuccessSuffix) {
+				continue
+			}
+
+			// Remove non-directory files
+			if err := os.Remove(packageTypeDirName); err != nil {
+				logger.Debugf("Failed to remove file %s: %v", packageTypeDirName, err)
+				allErrors = errors.Join(allErrors, err)
+			}
 			continue
 		}
-		// read all of the packages in the directory and delete those that aren't in expectedPackageEntries
+
+		// This is a package type directory (e.g., "module")
+		if runtime.GOOS == windowsOS {
+			if err := cleanupPackageTypeDirectoryWindows(logger, packageTypeDirName, expectedPackageEntries); err != nil {
+				allErrors = errors.Join(allErrors, err)
+			}
+		} else {
+			if err := cleanupPackageTypeDirectoryGeneric(logger, packageTypeDirName, expectedPackageEntries); err != nil {
+				allErrors = errors.Join(allErrors, err)
+			}
+		}
+
+		// After cleanup, check if package type directory is empty and can be removed
 		packageDirs, err := os.ReadDir(packageTypeDirName)
 		if err != nil {
+			logger.Debugf("Failed to read package type directory %s: %v", packageTypeDirName, err)
 			allErrors = errors.Join(allErrors, err)
 			continue
 		}
-		for _, entry := range packageDirs {
```

Comment:
i moved the previous code into its own function since i didn't want to touch the non windows code